app/utils/callback/lava.py

```diff
@@ -432,12 +433,14 @@ def _add_test_results(group, suite_results, suite_name):
     test_cases = []
     test_sets = {}
 
-    for test in tests:
+    for idx, test in enumerate(reversed(tests)):
```

Comment:
One issue with this is that the test case index goes through the whole test suite instead of being within each test set / group.  As a result, if you have 5 groups of 10 tests each, indexes in group 2 will start at 10.  So I think we could drop the `enumerate()` approach and do as I'm suggesting below:
```diff
diff --git a/app/utils/callback/lava.py b/app/utils/callback/lava.py
index a39af1dda5c5..e28fd6c6994a 100644
--- a/app/utils/callback/lava.py
+++ b/app/utils/callback/lava.py
@@ -449,6 +449,7 @@ def _add_test_results(group, suite_results, suite_name):
             test_case_list = test_sets.setdefault(test_set_name, [])
         else:
             test_case_list = test_cases
+        test_case[models.INDEX_KEY] = len(test_case_list)
         measurement = test.get("measurement")
         if measurement and measurement != 'None':
             test_case[models.MEASUREMENTS_KEY] = [{
```
Then I'm not sure if the index should start at 0 or 1, but I guess that's rather subjective.  Maybe starting at 1 is a bit more intuitive, in which case it would be `len(test_case_list) + 1`.

Reply:
Ok, it wasn't entirely clear to me if we want to have an index unique over the test suite or a test group. I'll change it then. I think it'd be better to have it starting from 1

Reply:
Changes applied

Reply:
Thanks, testing that again on staging now.